xbmc/utils/CryptThreading.cpp

```diff
@@ -57,11 +40,6 @@ CryptThreadingInitializer::CryptThreadingInitializer()
   // OpenSSL < 1.1 needs integration code to support multi-threading
   // This is absolutely required for libcurl if it uses the OpenSSL backend
   m_locks.resize(CRYPTO_num_locks());
-#if KODI_OPENSSL_USE_THREADID
-  CRYPTO_THREADID_set_callback(thread_id);
```

Comment:
`KODI_OPENSSL_USE_THREADID` == `OPENSSL_VERSION_NUMBER >= 0x10000000L`, which is always true for OpenSSL 1.0.2, so this line has to be kept.

Reply:
I don't think so. This is the issue brought up in the chat. If we're at 1.0.2 or later there's no need at all to manage custom thread_id. See https://www.openssl.org/docs/man1.0.2/man3/CRYPTO_THREADID_set_callback.html the paragraph about the thread id. it basically says: ```If the application does not register such a callback using CRYPTO_THREADID_set_callback(), then a default implementation is used - on Windows and BeOS this uses the system's default thread identifying APIs, and on all other platforms it uses the address of errno. The latter is satisfactory for thread-safety if and only if the platform has a thread-local error number facility.```

Reply:
I've only looked at the changes in this PR and now code differs from the one without this PR.
> [...], if the platform has a thread-local error number facility

Is this requirement fulfilled on all platforms (except Windows)?

Reply:
Some kodi components try to install their own openssl callback handlers, i.e. python or inputstream.adaptive. This may crash without kodi installing the handler first, see #15301.

Reply:
Okay. I'm done messing around with this. I'm just going to fix it so it works on all platforms the same no matter what. Thanks for the feedback guys. Once all platforms are on 1.1 we can delete this class all together.

Reply:
@mglae I rewrote it. Looking at the docs and the header files (on Ubuntu 18.04, which has OpenSSL >= 1.1.0) it looks like none of this functionality applies once you get to 1.1. Let me know if your experience with the previous problem you pointed out refutes this.

Reply:
@jimfcarroll LGTM

Yes, with OpenSSL >= 1.1.0 locking and ThreadID generation is done internally. #12351 is already waiting for Ubuntu 16.04 to die.